keanu-project/src/test/java/io/improbable/keanu/algorithms/mcmc/NetworkSamplesGeneratorTest.java

```diff
@@ -67,79 +66,80 @@ public void streamsExpectedNumberOfSamples() {
     }
 
     @Test
-    public void doesUpdateProgressAndFinishProgressOnGeneration() {
+    public void doesUpdateStatusAndFinishStatusOnGeneration() {
         AtomicInteger stepCount = new AtomicInteger(0);
         AtomicInteger sampleCount = new AtomicInteger(0);
 
-        ProgressBar progressBar = mock(ProgressBar.class);
+        StatusBar statusBar = mock(StatusBar.class);
         TestSamplingAlgorithm algorithm = new TestSamplingAlgorithm(stepCount, sampleCount);
 
-        NetworkSamplesGenerator unitUnderTest = new NetworkSamplesGenerator(algorithm, () -> progressBar);
+        NetworkSamplesGenerator unitUnderTest = new NetworkSamplesGenerator(algorithm, () -> statusBar);
         unitUnderTest.generate(10);
 
-        Mockito.verify(progressBar, times(10)).progress(anyString(), anyDouble());
-        Mockito.verify(progressBar).finish();
+        Mockito.verify(statusBar, times(1)).setMessage(anyString());
+        Mockito.verify(statusBar).finish();
     }
 
+
     @Test
-    public void doesCreateNewProgressBarOnGenerationFinish() {
+    public void doesCreateNewStatusBarOnGenerationFinish() {
         AtomicInteger stepCount = new AtomicInteger(0);
         AtomicInteger sampleCount = new AtomicInteger(0);
 
-        ProgressBar progressBar1 = mock(ProgressBar.class);
-        ProgressBar progressBar2 = mock(ProgressBar.class);
+        StatusBar statusBar1 = mock(StatusBar.class);
+        StatusBar statusBar2 = mock(StatusBar.class);
 
-        AtomicInteger progressBarCreationCount = new AtomicInteger(0);
+        AtomicInteger statusBarCreationCount = new AtomicInteger(0);
         TestSamplingAlgorithm algorithm = new TestSamplingAlgorithm(stepCount, sampleCount);
 
         NetworkSamplesGenerator unitUnderTest = new NetworkSamplesGenerator(algorithm, () -> {
-            int callNumber = progressBarCreationCount.getAndIncrement();
+            int callNumber = statusBarCreationCount.getAndIncrement();
             if (callNumber == 0) {
-                return progressBar1;
+                return statusBar1;
             } else {
-                return progressBar2;
+                return statusBar2;
             }
         });
 
         unitUnderTest.generate(10);
-        Mockito.verify(progressBar1, times(10)).progress(anyString(), anyDouble());
-        Mockito.verify(progressBar1).finish();
+        Mockito.verify(statusBar1, times(1)).setMessage(anyString());
```

Comment:
These are called less now because the `setMessage` only has to be done once. It's now the percentage component that updates.